SMLHelper/Patchers/PlayerPatcher.cs

```diff
@@ -0,0 +1,53 @@
+﻿using HarmonyLib;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace SMLHelper.V2.Patchers
+{
+    internal class PlayerPatcher
+    {
+            [HarmonyPatch(typeof(Player), nameof(Player.Awake))]
+            [PatchUtils.Postfix]
+            internal static void Player_Awake_Postfix(Player __instance)
+            {
+                __instance.gameObject.EnsureComponent<FloatingOrigin>().ReferenceObject = __instance.transform;
+            }
+            [HarmonyPatch(typeof(Player), nameof(Player.SetPosition), new Type[] { typeof(Vector3) })]
+            [PatchUtils.Prefix]
+            internal static bool Player_SetPosition_Prefix(ref Vector3 wsPos)
+            {
+                wsPos -= FloatingOrigin.CurrentOffset;
+                return true;
+            }
+            [HarmonyPatch(typeof(Player), nameof(Player.UpdateBiomeRichPresence))]
+            [PatchUtils.Postfix]
+            public static void Player_UBRP_Postfix(string biomeStr)
+            {
```

Comment:
This patch should be removed since every biome should already have an atmosphere volume and that will take care of the RP things.

Reply:
I also want people to be able to have custom rich presences.

Reply:
That can be done by adding a language line. the key should be `PresenceExploring_biome_{BIOME_NAME}` and the game will automatically set its value as RP when the player is there.

Reply:
Ah, alright.